### ISSUE TITLE
Significant Performance Degradation on Large File

### ISSUE THREAD

**Issue by pingshunhuangalex** [VSCodeVim/Vim#8115]
**Describe the bug**
I have a couple of 120k lines YAML files which I edit on a daily basis. With recent updates (in the past month or even a couple of weeks ago), VS Code has become so laggy that the letter needs to catch up when I type, and moving lines up and down becomes a nightmare. Keystrokes also get swallowed from time to time which makes things unusable.

Turning the extension off seems to fix the issue, but I really don't want to. Also please note that this was never a problem with VS Code Vim until recently. I can reproduce it on both Windows and Mac machines. Tried fresh reinstallation, but it doesn't do anything.

**To Reproduce**

1.  Open a large file with let's say 120k
2.  Type quickly or try any basic line movement

**Expected behavior**
There shouldn't be any lag for basic operations and typing

**Screenshots**
*With VS Code Vim ON*

![Nov-24-2022 10-46-34](https://user-images.githubusercontent.com/17170199/203664756-6376023e-8f0f-4069-b31f-841afbab496d.gif)

*With VS Code Vim OFF*

![Nov-24-2022 10-42-58](https://user-images.githubusercontent.com/17170199/203664426-462c06b6-62a5-4cb5-93d4-9efc397550cd.gif)

**Environment (please complete the following information):**

```
Version: 1.74.0-insider
Date: 2022-11-23T05:28:16.879Z
Electron: 19.1.3
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Darwin arm64 21.6.0
Sandboxed: Yes
VS Code Vim Version: v1.24.3
```

**Comment by Bananaman** [comment]
@pingshunhuangalex Interesting problem.

Have you tried this instead?

https://marketplace.visualstudio.com/items?itemName=asvetliakov.vscode-neovim

According to this article, it's much, much faster:

https://galenwong.github.io/blog/2021-03-22-vscode-neovim-vs-vscodevim/

And it uses native NeoVim, so there's no need to emulate Vim in JavaScript. It's all running natively with full support for all Vim commands.

**Comment by pingshunhuangalex** [comment]
Thanks for the suggestion @Bananaman 

I've hesitated before between the 2 options and picked VSCodeVim back then, probably cuz it's maintained by Microsoft devs.

I've accumulated quite a bit of configs and became reluctant to migrate. Maybe it's time to change now. Thanks.

**Comment by Bananaman** [comment]
@pingshunhuangalex Yeah I know the problem about having a lot of configs, but I think it's better to have a usable editor than a super laggy one. Slow text is so bad for productivity and peace of mind. :/

I heard so many good reviews for NeoVim plugin and will switch to it.

VSCodeVim isn't maintained by Microsoft by the way? At least if I look at the contributors, I can't find any Microsoft devs:

https://github.com/VSCodeVim/Vim/graphs/contributors

It's mainly maintained by https://github.com/J-Fields who works at another company.

**Comment by Bananaman** [comment]
@pingshunhuangalex Oh, I was thinking about this again and just realized that there may have been a misunderstanding.

I wasn't saying "switch to NeoVim". :D

- VSCode: Owned and maintained by microsoft.
- VSCodeVim: This plugin, tries to emulate Vim in JavaScript but is slow because of VSCode issues.
- NeoVim: A different editor that uses different configs. I don't want to switch to it, because it's hard to set up and uses 100% keyboard controls, no GUI. I have used Vim in terminals in the past as my main editor and it's good but it's so much work to set up and maintain it. So I am not gonna go back to it.
- NeoVim Plugin for VSCode: It's a plugin that makes VSCode use a NeoVim "process" for text input which means 100% perfect Vim compatibility inside VSCode.

The last one is what I said we should switch to:

https://marketplace.visualstudio.com/items?itemName=asvetliakov.vscode-neovim

**Comment by MurtadhaInit** [comment]
Thank you for the suggestion @Bananaman I didn't know such an extension existed. I'll give it a try. I came across this issue as the first thing I've noticed immediately after installing this plugin is the bizarre inconsistency even when navigating in normal mode.

Sometimes hitting the h key moves left, and sometimes it just decides not to do anything! Same thing when holding it: it either moves normally as expected, or just takes a moment then moves.

It's a shame. I really hope Microsoft works on a native Vim plugin, even if with a basic functionality. I think the Vim plugin in JetBrains IDEs is developed by JetBrains.

**Comment by jvanheugten** [comment]
I have the same issue but in a 6000 line Markdown file. Typing gets really slow with the VIM extension enabled (I had found this through the VSCode Bisection of plugins).

**Comment by LuSrackhall** [comment]
> 哦，我又想了想，才意识到可能有误会。
> 
> 我不是说 “切换到 NeoVim”。:D
> 
> * VSCode：由 Microsoft 拥有和维护。
> * VSCodeVim：此插件尝试在 JavaScript 中模拟 Vim，但由于 VSCode 问题而运行缓慢。它不是由 Microsoft 维护的。
> * NeoVim：使用不同配置的不同编辑器。我不想切换到它，因为它很难设置并且使用 100% 的键盘控制，没有 GUI。我过去在终端中使用 Vim 作为我的主要编辑器，这很好，但设置和维护它需要做很多工作。所以我不打算回到它。
> * VSCode 的 NeoVim 插件：这是一个插件，它使 VSCode 使用 NeoVim“进程”进行文本输入，这意味着 VSCode 内部 100% 完美的 Vim 兼容性。它甚至支持 Vim 插件，因此我们可以添加诸如 “” 之类的内容，用于 “删除括号内” 插件和 EasyMotion 插件等。它使用 VSCode 进行“插入模式”（文本输入/自动完成），并使用 NeoVim 进行移动和编辑移动（如跳跃/移动、删除、复制、粘贴），每次按 Escape 退出“插入模式”时。所以这是两全其美的。它甚至支持高级 Vim作，例如可视模式的块复制和块粘贴。`di{`
> 
> 最后一个是我说的我们应该切换到：
> 
> https://marketplace.visualstudio.com/items?itemName=asvetliakov.vscode-neovim


I noticed that in the settings of VSCodeVim, it seems to support using Neovim as a backend and can also use some .vimrc configurations. I wonder if this could solve your problems.
>> https://github.com/VSCodeVim/Vim/issues/1735
>> https://github.com/VSCodeVim/Vim/pull/1725
>> https://github.com/VSCodeVim/Vim/issues/1724
>>
>>> ![Image](https://github.com/user-attachments/assets/a58a3ef3-9b50-4efc-af14-0321ccadae06)
